### PR TITLE
[Set-DbaAgentAlert] - Added ShouldProcess to the Rename to fix #8137

### LINKED ISSUE #8137: Set-DbaAgentAlert - renames an alert even with Whatif
### Verified issue does not already exist?

Yes

### What error did you receive?

No error, Alert was renamed when it shouldn't have been.

### Steps to Reproduce

```powershell
Set-DbaAgentAlert -SqlInstance $s1 -Alert dbatoolsci_test_1062943236 -NewName dbatoolsci_dontchange -WhatIf
Get-DbaAgentAlert -SqlInstance $s1 -Alert dbatoolsci_dontchange
```

### Are you running the latest release?

Yes

### Other details or mentions

Similar to #8066

### What PowerShell host was used when producing this error

Windows PowerShell (powershell.exe), Windows PowerShell ISE (powershell_ise.exe)

### PowerShell Host Version

Name                           Value                                                                            
----                           -----                                                                            
PSVersion                      5.1.19041.1320                                                                   
PSEdition                      Desktop                                                                          
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                          
BuildVersion                   10.0.19041.1320                                                                  
CLRVersion                     4.0.30319.42000                                                                  
WSManStackVersion              3.0                                                                              
PSRemotingProtocolVersion      2.3                                                                              
SerializationVersion           1.1.0.1                                                                          

### SQL Server Edition and Build number

2019

### .NET Framework Version

.NET Framework 4.8.4470.0

### PR DESCRIPTION
**Pull request by MikeyBronowski**
## Type of Change

 - [x] Bug fix (non-breaking change, fixes #8137 )
 - [ ] New feature (non-breaking change, adds functionality, fixes # )
 - [ ] Breaking change (effects multiple commands or functionality, fixes # )
 - [x] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1`)
 - [ ] Adding code coverage to existing functionality
 - [ ] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/dataplat/appveyor-lab ?
 - [ ] Unit test is included
 - [ ] Documentation
 - [ ] Build system

### CODE REVIEW

File: functions/Set-DbaAgentAlert.ps1 [hunk 1]
```diff
@@ -132,8 +132,9 @@ function Set-DbaAgentAlert {
             #region alert options
             # Settings the options for the alert
             if ($NewName) {
-                Write-Message -Message "Setting alert name to $NewName" -Level Verbose
-                $currentAlert.Rename($NewName)
+                if ($Pscmdlet.ShouldProcess($server, "Setting alert name to $NewName for $a")) {
+                    $JobSchedule.Rename($NewName)
```
Comment (andreasjordan) [inline functions/Set-DbaAgentAlert.ps1:136]: `$JobSchedule`?

File: functions/Set-DbaAgentAlert.ps1 [hunk 2]
```diff
@@ -149,16 +150,16 @@ function Set-DbaAgentAlert {
...
-                    Write-Message -Message "Changing the alert" -Level Verbose
+                    Write-Message -Message "Committing changes for alert $a" -Level Verbose
 
                     # Change the alert
                     $currentAlert.Alter()
                 } catch {
                     Stop-Function -Message "Something went wrong changing the alert" -ErrorRecord $_ -Target $instance -Continue
                 }
-                Get-DbaAgentAlert -SqlInstance $server | Where-Object Name -eq $currentAlert.name
+                Get-DbaAgentAlert -SqlInstance $server | Where-Object Name -EQ $currentAlert.name
```
Comment (andreasjordan) [inline functions/Set-DbaAgentAlert.ps1:162]: Even if not the correct case, we use `-eq` with `Where-Object` for better readability.

File: functions/Set-DbaAgentAlert.ps1 [hunk 3]
```diff
@@ -132,7 +132,7 @@ function Set-DbaAgentAlert {
             #region alert options
             # Settings the options for the alert
             if ($NewName) {
-                if ($Pscmdlet.ShouldProcess($server, "Setting alert name to $NewName for $Alert")) {
+                if ($Pscmdlet.ShouldProcess($server, "Setting alert name to $NewName for $a")) {
```
Comment (andreasjordan) [inline functions/Set-DbaAgentAlert.ps1:135]: Why have you changed that? $a is only the last alert from the foreach loop. Shouldn't it be `$currentAlert`?

File: functions/Set-DbaAgentAlert.ps1 (hunk 2 above)
Comment (MikeyBronowski) [reply functions/Set-DbaAgentAlert.ps1:162]: Sorry, don't follow - do you want me to change here something?

Comment (MikeyBronowski) [reply functions/Set-DbaAgentAlert.ps1]: Gotcha. I caught it in the other command, but this was changed by the formatted and I missed it to change it back to `-eq`

File: functions/Set-DbaAgentAlert.ps1 [hunk 4]
```diff
@@ -149,9 +150,9 @@ function Set-DbaAgentAlert {
             #endregion alert options
 
             # Execute
-            if ($PSCmdlet.ShouldProcess($SqlInstance, "Changing the alert $a")) {
+            if ($PSCmdlet.ShouldProcess($SqlInstance, "Committing changes for alert $a")) {
```
Comment (andreasjordan) [inline functions/Set-DbaAgentAlert.ps1:153]: Like in line 135, this should not be `$a` but `$currentAlert`.

File: functions/Set-DbaAgentAlert.ps1 [hunk 5]
```diff
@@ -149,9 +150,9 @@ function Set-DbaAgentAlert {
...
                 try {
-                    Write-Message -Message "Changing the alert" -Level Verbose
+                    Write-Message -Message "Committing changes for alert $a" -Level Verbose
```
Comment (andreasjordan) [inline functions/Set-DbaAgentAlert.ps1:155]: You don't need this line, as ShouldProcess also has verbose output. (But if you still like to have the line: Like in line 135, this should not be `$a` but `$currentAlert`.)

**Comment by potatoqualitee** [comment]
I will hold off until this is approved ✅

**Comment by potatoqualitee** [comment]
I take that back, we can fix those small issues in another PR, this is an important fix! thank you both 🙇🏼

**Pull request merged**